File: server/internal/helpers/validation.go
package helpers

import (
	"github.com/Shiwang0-0/HLS-Transcoder/server/internal/models"
	"github.com/gofiber/fiber/v2"
)

func validationError(msg string) *fiber.Error {
	return &fiber.Error{
		Code:    fiber.StatusBadRequest,
		Message: msg,
	}
}

func ValidateData(data models.InitMultipartUploadRequest) *fiber.Error {
	allowedTypes := map[string]bool{
		"video/mp4": true,
	}

	if !allowedTypes[data.Type] {
		return validationError("Media type not allowed")
	}

	const maxSize = 500 * 1024 * 1024
	if data.Size > maxSize {
		return validationError("File size cannot exceed 500MB")
	}

	if data.Name == "" {
		return validationError("File name is required")
	}

	if data.Size <= 0 {
		return validationError("Invalid file size")
	}

	if data.Fingerprint == "" {
		return validationError("Fingerprint is required")
	}

	return nil
}

File: client/src/helpers/validation.js
export function validateFile(file) {
    const allowedTypes = ["video/mp4"]
    const maxSize = 500 * 1024 * 1024 // 500 MB

    if (!file?.name?.trim()) {
        return "File name is required"
    }

    if (!allowedTypes.includes(file.type)) {
        return "Media type not allowed, try .mp4"
    }

    if (file.size <= 0) {
        return "Invalid file size"
    }

    if (file.size > maxSize) {
        return "File size cannot exceed 500MB"
    }

    return null
}